src/media-event-filter.ts

```diff
@@ -133,8 +137,12 @@ export const getMediaEventFilter = ({
   };
 
   const onCanPlay = (): void => {
-    if (isNotReady()) return;
-
+    if (isNotReady()) {
+      if (mp4Mode) {
+        onCanPlayThrough();
+      }
+      return;
+    }
```

Comment:
NIT:

```suggestion
    if (mp4Mode && isNotReady()) {
      onCanPlayThrough();
      return;
    }
```

technically saves some CPU cycles, but I also prefer the syntax to nested if's